src/libs/pronouns.js

```diff
@@ -0,0 +1,232 @@
+	const resp = await fetch(`https://en.pronouns.page/api/profile/get/${username}?version=2`);
+	if (resp.status >= 400) {
+		return null;
+	}
+
+	const { profiles } = await resp.json();
+	if (!profiles) return null;
+
+	// Unfortunately, pronouns.page does not return a 404 if a profile does not exist, but an empty profiles object. :clown_face:
+	if (!Object.keys(profiles).length) return null;
+
+	let pronouns;
+	// Query the pronouns in the following language order:
+	// 1. The mastodon interface language
+	// 2. The spoken languages according to the user
+	// 3. The english language.
+	const languages = [document.documentElement.lang, ...window.navigator.languages, "en"];
+	for (const lang of languages) {
+		if (lang in profiles) {
+			pronouns = profiles[lang].pronouns;
+			break;
+		}
+	}
+
+	// If we don't have a value yet, just take the first profile.
+	if (!pronouns) pronouns = profiles[0].pronouns;
+
+	let val = pronouns.find((x) => x.opinion === "yes").value;
+	val = sanitizePronounPageValue(val);
+	return val;
+}
+
+/**
+ * @param {string} val
+ */
+function sanitizePronounPageValue(val) {
+	if (!val.startsWith("https://")) return val;
+
+	val = val.replace(/https?:\/\/.+\.pronouns\.page\/:?/, "");
+
+	if (val === "no-pronouns") val = "no pronouns";
+	return val;
+}
+
+const knownPronouns = [
+	"ae",
+	"aer",
+	"aers",
+	"aerself",
+	"co",
+	"co's",
+	"cos",
+	"coself",
+	"e",
+	"eir",
+	"eirs",
+	"em",
+	"ems",
+	"emself",
+	"es",
+	"ey",
+	"fae",
+	"faer",
+	"faers",
+	"faerself",
+	"he",
+	"her",
+	"hers",
+	"herself",
+	"him",
+	"himself",
+	"hir",
+	"hirs",
+	"hirself",
+	"his",
+	"hu",
+	"hum",
+	"hus",
+	"huself",
+	"it",
+	"its",
+	"itself",
+	"ne",
+	"nem",
+	"nemself",
+	"nir",
+	"nirs",
+	"nirself",
+	"one",
+	"one's",
+	"oneself",
+	"per",
+	"pers",
+	"perself",
+	"s/he",
+	"she",
+	"their",
+	"theirs",
+	"them",
+	"themself",
+	"themselves",
+	"they",
+	"thon",
+	"thon's",
+	"thons",
+	"thonself",
+	"ve",
+	"ver",
+	"vers",
+	"verself",
+	"vi",
+	"vim",
+	"vims",
+	"vimself",
+	"vir",
+	"virs",
+	"virself",
+	"vis",
+	"xe",
+	"xem",
+	"xemself",
+	"xyr",
+	"xyrs",
+	"ze",
+	"zhe",
+	"zher",
+	"zhers",
+	"zherself",
+	"zir",
+	"zirs",
+	"zirself",
+];
+
+/**
+ * Tries to extract pronouns from the bio/note. Only "known" pronouns are returned, which is
+ * a compromise for the pattern matching. At no point we want to limit the pronouns used by persons.
+ * @param {string} bio The bio
+ * @returns {string|null} The result or null
+ */
+function extractFromBio(bio) {
+	const exactMatches = bio.matchAll(/(\w+) ?\/ ?(\w+)/gi);
+	for (const [match, subjective, objective] of exactMatches) {
+		if (knownPronouns.includes(subjective) && knownPronouns.includes(objective)) {
+			return match.replaceAll(" ", "");
+		}
+	}
+
+	const followedByColon = bio.matchAll(/pronouns?:\W+([\w/+]+)/gi);
+	for (const match of followedByColon) {
+		return match.pop(); // first group is last entry in array
+	}
+
```

Comment:
it looks like we're missing out on "any/all" or "any pronouns".

how about another matcher like `/[any|all]+ pronouns/gi`?

Reply:
Added those.